app/Jobs/GenerateHashValue.php

```diff
@@ -75,61 +75,62 @@ public function __construct(Image $image)
      * Execute the job.
      *
      * @return void
+     * @throws Exception
      */
     public function handle()
     {
-        try {
-            if ($this->needsHashValue($this->image)) {
-                FileCache::getOnce($this->image, function ($image, $path) {
-                    if (!File::exists($path)) {
-                        throw new Exception("File '{$path}' does not exist.");
-                    }
-                    if ($this->needsHashValue($image)) {
-                        $this->getThumbnail($image, $path);
-                        // here path or the image which i get with get Thumbnail?
-                        $output = $this->python("{$path}");
-                        $image->hash = $output;
-                        $image->save();
-                    }
-
-                });
-            }
-        } catch (Exception $e) {
-            if (App::runningUnitTests()) {
-                throw $e;
-            } else {
-                Log::warning("Could not process new image {$this->image->id}: {$e->getMessage()}");
-            }
-        }
+        $hashValue = $this->getHashValue($this->image);
+        $id = $hashValue->id;
+        $hash = $hashValue->hash;
+
+        dd($hash);
+        $this->image->hash = $hash;
+        $this->image.save();
+
     }
 
     /**
-     * Chack if an image needs a hash.
+     * Execute the HashValueGenerator and get the resulting hash value to the image.
      *
      * @param Image $image
-     *
-     * @return bool
+     * @param $path
+     * @return string
+     * @throws Exception
      */
-    protected function needsHashValue(Image $image)
+    protected function getHashValue(Image $image)
     {
-        if (!array_key_exists($image->id, $this->needsHashCache)) {
-            $prefix = fragment_uuid_path($image->uuid);
-            $format = config('thumbnails.format');
-            $this->needsHashCache[$image->id] =
-                !Storage::disk(config('thumbnails.storage_disk'))
-                    ->exists("{$prefix}.{$format}");
-        }
+        return FileCache::get($image->image, function ($image, $path) use ($image) {
```

Comment:
I am not sure if I get the image and his path for creating a temporary path. 
I also removed get Thumbnail because it looks like it is not necessary anymore. Is this correct @mzur

Reply:
You don't need to use the FileCache. This retrieves the original image file but you only need the thumbnail. To get the thumbnail (as byte string) you can do [this](https://github.com/biigle/core/blob/0e8481d373c4415a7bc45769cded45505aab4dc8/app/Jobs/ProcessNewImage.php#L192-L193) (only with `get ` instead of `put`). As to temporary files, see my last email regarding `File::put`.